### PR TITLE
Add `match-default-export-name` rule

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [ ] Addresses an existing issue: #0000
- [X] New feature, bugfix, or enhancement
  - [X] Includes tests
- [X] Documentation update

#### What changes did you make?

Added the `match-default-export-name` rule, which requires that default imports have the same name as the default exports they reference.
This doesn't check named exports, since they require explicit `import { foo as bar }` and are unlikely to be name-changed by mistake.
This also doesn't check `import =` and `export =`, since `export =` comes from declaration files which probably don't use the name you want, and if it's an `@types` module you can't simply change it.

### CODE REVIEW

**Comment by adidahiya** [comment]
very cool! this programming error was one of my main motivations for the [no-default-export](https://palantir.github.io/tslint/rules/no-default-export/) rule.

File: src/rules/matchDefaultExportNameRule.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.AbstractRule {
```
Comment (nchen63) [inline src/rules/matchDefaultExportNameRule.ts]: use `TypedRule` and `applyWithProgram`

File: test/rules/match-default-export-name/test.ts.lint [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import b from "./named";
+       ~ [Default export 'a' is imported as 'b'.]
+import anyName from "./anonymous";
```
Comment (andy-hanson) [inline test/rules/match-default-export-name/test.ts.lint]: That's not legal syntax anyway.

**Comment by nchen63** [comment]
@andy-hanson thanks!

**Pull request merged**